src/app/modules/examples/examples.utils.ts

```diff
@@ -0,0 +1,224 @@
+    lookUpMiddleware,
+    // Only display forms with more than minSubmissionCount submissions.
+    filterBySubmissionCount(minSubmissionCount),
+    // Sort by how well search terms were matched.
+    sortByRelevance,
+    // Retrieve form feedback from the forms that reach this step.
+    lookupFormFeedback,
+  )
+}
+
+/**
+ * Creates a query pipeline that can be used to retrieve forms for the /examples
+ * page.
+ *
+ * This query will return forms sorted by last submitted date, filtered by
+ * number of submissions (greater than given minSubmissionCount), public forms,
+ * and selected agency.
+ * @param groupByMiddleware The inisital The id of the agency to filter forms listed by. If no id is given, all matched forms will be returned
+ * @param agencyId Optional. The id of the agency to filter forms listed by. If no id is given, all matched forms will be returned
+ * @param minSubmissionCount The minimum submission count returned forms must have to be returned by the aggregation pipeline
+ */
+export const createGeneralQueryPipeline = ({
+  groupByMiddleware,
+  agencyId,
+  minSubmissionCount,
+}: {
+  groupByMiddleware: Record<string, unknown>[]
+  agencyId?: string
+  minSubmissionCount: number
+}): Record<string, unknown>[] => {
+  return groupByMiddleware.concat(
+    // Only display forms with more than minSubmissionCount submissions.
+    filterBySubmissionCount(minSubmissionCount),
+    // More recently submitted forms appear higher on the examples page.
+    sortByLastSubmitted,
+    // Retrieve form info from the forms that reach this step.
+    lookupFormInfo,
+    // Filter out all inactive/unlisted forms.
+    filterInactiveAndUnlistedForms,
+    // Retrieve agency infos of forms in this stage.
+    lookupAgencyInfo,
+    // Filter by agency id if parameter given.
+    agencyId ? filterByAgencyId(agencyId) : [],
+    // Retrieve form feedback from the forms that reach this step.
+    lookupFormFeedback,
+  )
+}
+
+/**
+ * Creates a query pipeline that can be used to retrieve the example card data
+ * for the /examples page with the specified form id.
+ * This pipeline only provides formId, form title, agency logo and color theme.
+ * @param formId The specific form to query.
+ */
+export const createFormIdInfoPipeline = (
+  formId: string,
+): Record<string, unknown>[] => {
+  return searchFormsById(formId).concat(
+    filterInactiveAndUnlistedForms,
+    lookupAgencyInfo,
+    projectFormDetails,
+  )
+}
+
+/**
+ * Creates a query pipeline that can be used to retrieve a single example form
+ * for the /examples page using the submission collection.
+ *
+ * This pipeline will return the average feedback for the form id referenced to
+ * be shown in the example form.
+ * @param formId. The id of the form to retrieve data for
+ */
+export const createSingleSearchSubmissionPipeline = (
+  formId: string,
+): Record<string, unknown>[] => {
+  return searchSubmissionsForForm('form', formId).concat(
```

Comment:
add a comment why the key is `form`?

Reply:
`form` is the "foreign key" in the submissions collection

Reply:
Actually, we just need to emphasise what's the difference in the 2 calls to `searchSubmissionsForForm`